### ISSUE TITLE
Order create: OrderLineData TypeScript Interface marks optional parameters as required

### ISSUE THREAD

**Issue by mscholtus** [mollie/mollie-api-node#157]
When creating an order with `mollieClient.orders.create` using the provided Typescript example code, Typescript shows errors indicating that several parameters are missing. These parameters are not required for creating an order but these parameters are marked as required by the `OrderLineData` Interface provided by `@mollie/api-node`.

I use the code provided in the example: https://github.com/mollie/mollie-api-node/blob/master/examples/orders/create.ts

The example functions fine, but my editor and build steps throw errors due to the Typescript problems.
I have the add the following dummy parameters before the Typescript errors go away:

```
          id: '',
          isCancelable: false,
          amountShipped: {
            currency: 'EUR',
            value: '0',
          },
          amountCanceled: {
            currency: 'EUR',
            value: '0',
          },
          amountRefunded: {
            currency: 'EUR',
            value: '0',
          },
          quantityCanceled: 0,
          quantityRefunded: 0,
          quantityShipped: 0,
          shippableQuantity: 0,
          refundableQuantity: 0,
          cancelableQuantity: 0,
          createdAt: 'test',
          metadata: {},
          _links: {
            productUrl: {
              href: '',
              type: '',
            },
            imageUrl: {
              href: '',
              type: '',
            },
          },
          resource: 'orderline',
          status: OrderStatus.created,
```

**Comment by Pimm** [comment]
Thanks for brining this up, Marijn! I'll look into it.

**Comment by mscholtus** [comment]
@Pimm Great, thanks. I think this issue actually is present across all calls when using TypeScript. The Interface declares a response object coming from the Mollie API, and that data is valid. But not all those parameters are required for an API request to be valid. My suggestion would be to introduce a request object Interface to differentiate the required parameters when calling the API from the API response.

I’ve tried making parameters optional in the `OrderLine.d.ts` but then the `resource` and `id` parameters stemming from the lower level `Model.d.ts` still pop up. So that quick fix doesn't work and also wouldn't scale well.

**Comment by Pimm** [comment]
It is not the case that properties in the objects coming _from_ the Mollie server are by definition required in the objects going _to_ the Mollie server. We do make that distinction between mandatory and optional properties internally. Most likely we just made a mistake somewhere.

I will try to look into this today.

**Comment by Pimm** [comment]
A version which fixes this should be released soon.

If that is not fast enough, you can of course always opt out of TS type safety:
```javascript
const order = await (mollieClient.orders.create as (parameters: any) => Promise<Order>)({
  
});
```

**Comment by vernondegoede** [comment]
We just published version 3.2.2, which includes the fix for this issue.

Thanks!

**Comment by mscholtus** [comment]
The issue is fixed, thanks for your efforts @Pimm @vernondegoede!
